src/components-styled/sidebar-metric/sidebar-metric.tsx

```diff
@@ -0,0 +1,110 @@
+import { assert } from '~/utils/assert';
+import { get } from 'lodash';
+import { isDefined } from 'ts-is-present';
+import { Box } from '~/components-styled/base';
+import { SidebarBarScale } from './sidebar-barscale';
+import { SidebarKpiValue } from './sidebar-kpi-value';
+import { replaceVariablesInText } from '~/utils/replaceVariablesInText';
+import { formatDateFromSeconds } from '~/utils/formatDate';
+import siteText from '~/locale/index';
+import { getDataConfig, DataScope } from './data-config';
+
+interface SidebarMetricProps<T> {
+  scope: DataScope;
+  data: T;
+  metricName: string;
+  metricProperty: string;
+  localeTextKey: string;
+  differenceKey?: string;
+  showBarScale?: boolean;
+  isWeeklyData?: boolean;
+  annotationKey?: string;
+}
+
+export function SidebarMetric<T>({
+  scope,
+  data,
+  metricName,
+  metricProperty,
+  localeTextKey,
+  differenceKey,
+  showBarScale,
+  annotationKey,
+}: SidebarMetricProps<T>) {
+  const lastValue = get(data, [metricName, 'last_value']);
```

Comment:
What's wrong with vanilla js?
```suggestion
  const lastValue = data[metricName].last_value;
```

Reply:
Of you are not sure that `data[metricName]` exists it can generate a runtime exception.

Reply:
yes, which still happens due to the assertion on the next few lines ;-)

Reply:
The assertion won't happen if this line tries to index from undefined. And then you will not get a nice message like in the assertion